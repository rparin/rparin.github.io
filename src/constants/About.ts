const AboutGen = {
  Initials: "RP",
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, Web Developer, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels",
  "playing games",
  "watching movies",
  "learning new things",
];

const Socials = {
  github: "https://github.com/rparin",
  linkedIn: "https://www.linkedin.com/in/rparin",
};

const AboutMe = {
  Intro: "A little about me: my hobbies, interests, and goals.",

  Goals:
    "The thrill of crafting with my hands led me to computer science, where I aspire to make a meaningful difference in the lives of others, whether through a useful digital tool or a fun game.",
  GoalsSm:
    "Crafting drove me to computer science to make a difference, via games or digital tools.",

  Hobbies:
    "I cherish stories for the unique lessons they offer, particularly the no-happy-ending tales by my favorite author, Sugaru Miaki. They've taught me to embrace failure as a path to improvement.",
  HobbiesSm:
    "I enjoy reading for the lessons they offer. They've, taught me to overcome my regrets.",

  Interests:
    "One of my passion lies in creating memorable moments with friendsâ€”whether it's trying new flavors at local eateries, embarking on food adventures, or exploring exciting new places together.",
  InterestsSm:
    "One hobby of mine includes trying new foods and sightseeing with friends.",

  Gaming:
    "Gaming with my friends has become my digital backyard, as life takes us on different paths and distances grow, this space keeps us connected.",
  GamingSm:
    "Gaming with friends is my digital backyard, keeping me close despite life's changes.",
};

export { AboutGen, AboutSent, Socials, AboutMe };
