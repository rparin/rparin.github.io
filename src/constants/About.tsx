const AboutGen = {
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, WebDeveloper, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels.",
  "playing games.",
  "watching movies.",
  "trying new things.",
];

export { AboutGen, AboutSent };
