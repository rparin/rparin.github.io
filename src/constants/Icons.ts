import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";
import zod from "@icons/Zod.svg";
import springboot from "@icons/SpringBoot.svg";
import docker from "@icons/Docker.svg";

type Icon = {
  name: string;
  img: StaticImageData;
  alt: string;
  invert?: boolean;
  url?: string;
};

const SocialIcons = {
  github: {
    name: "Github",
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    name: "LinkedIn",
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    name: "Gmail",
    img: gmail,
    alt: "logo of Gmail",
  },
};

const DevTools = {
  angular: {
    name: "Angular",
    img: angular,
    alt: "logo of angular",
    url: "https://angular.io/",
  },
  react: {
    name: "React",
    img: react,
    alt: "logo of react",
    url: "https://react.dev/",
  },
  nextjs: {
    name: "Nextjs",
    img: nextjs,
    alt: "logo of nextjs",
    url: "https://nextjs.org/",
  },
  springboot: {
    name: "Spring Boot",
    img: springboot,
    alt: "logo of Spring Boot",
    url: "https://spring.io/projects/spring-boot",
  },
  reactquery: {
    name: "Tanstack Query",
    img: reactquery,
    alt: "logo of reactquery",
    url: "https://tanstack.com/query/latest",
  },
  zod: {
    name: "Zod",
    img: zod,
    alt: "logo of zod",
    url: "https://zod.dev",
  },
  docker: {
    name: "Docker",
    img: docker,
    alt: "logo of Docker",
    url: "https://www.docker.com",
  },
  mapbox: {
    name: "Mapbox",
    img: mapbox,
    alt: "logo of mapbox",
    url: "https://www.mapbox.com/",
  },
  ionic: {
    name: "Ionic",
    img: ionic,
    alt: "logo of ionic",
    url: "https://ionicframework.com/",
  },
  tailwind: {
    name: "Tailwind",
    img: tailwind,
    alt: "logo of tailwind",
    url: "https://tailwindcss.com/",
  },
  fbase: {
    name: "Firebase",
    img: fbase,
    alt: "logo of firebase",
    url: "https://firebase.google.com/",
  },
  nodejs: {
    name: "Nodejs",
    img: nodejs,
    alt: "logo of nodejs",
    url: "https://nodejs.org/en",
  },
  cpp: {
    name: "C++",
    img: cpp,
    alt: "logo of c++",
    url: "https://isocpp.org/",
  },
  css: {
    name: "CSS",
    img: css,
    alt: "logo of css",
    url: "https://developer.mozilla.org/en-US/docs/Web/CSS",
  },
  js: {
    name: "Javascript",
    img: js,
    alt: "logo of javascript",
    url: "https://developer.mozilla.org/en-US/docs/Web/javascript",
  },
  html: {
    name: "HTML",
    img: html,
    alt: "logo of html",
    url: "https://developer.mozilla.org/en-US/docs/Web/HTML",
  },
  ts: {
    name: "Typescript",
    img: ts,
    alt: "logo of typescript",
    url: "https://www.typescriptlang.org/",
  },
  python: {
    name: "Python",
    img: python,
    alt: "logo of python programming language",
    url: "https://www.python.org/",
  },
  java: {
    name: "Java",
    img: java,
    alt: "logo of java",
    url: "https://www.java.com/en/download/help/whatis_java.html",
  },
  pygame: {
    name: "Pygame",
    img: pygame,
    alt: "logo of pygame library",
    url: "https://www.pygame.org/",
  },
  sfml: {
    name: "SFML",
    img: sfml,
    alt: "logo of SFML library",
    url: "https://www.sfml-dev.org/",
  },
  tensflow: {
    name: "Tensorflow",
    img: tensflow,
    alt: "logo of tensorflow",
    url: "https://www.tensorflow.org/",
  },
  pandas: {
    name: "Pandas",
    img: pandas,
    alt: "logo of pandas library",
    url: "https://pandas.pydata.org/",
  },
  flask: {
    name: "Flask",
    img: flask,
    alt: "logo of flask framework",
    url: "https://flask.palletsprojects.com/en/3.0.x/",
  },
  openai: {
    name: "OpenAi",
    img: openai,
    alt: "logo of openai",
    url: "https://openai.com/",
  },
};

export { DevTools, SocialIcons };
export type { Icon };
