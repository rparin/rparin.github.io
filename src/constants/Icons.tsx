import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import download from "@icons/Download.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";

type Img = {
  img: StaticImageData;
  alt: string;
  invert?: boolean;
};

const MiscIcons = {
  download: {
    img: download,
    alt: "download icon",
  },
};

const SocialIcons = {
  github: {
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    img: gmail,
    alt: "logo of Gmail",
  },
};

const CodeIcons = {
  angular: {
    img: angular,
    alt: "logo of angular",
  },
  react: {
    img: react,
    alt: "logo of react",
  },
  nextjs: {
    img: nextjs,
    alt: "logo of nextjs",
  },
  reactquery: {
    img: reactquery,
    alt: "logo of reactquery",
  },
  mapbox: {
    img: mapbox,
    alt: "logo of mapbox",
  },
  ionic: {
    img: ionic,
    alt: "logo of ionic",
  },
  tailwind: {
    img: tailwind,
    alt: "logo of tailwind",
  },
  fbase: {
    img: fbase,
    alt: "logo of firebase",
  },
  nodejs: {
    img: nodejs,
    alt: "logo of nodejs",
  },
  cpp: {
    img: cpp,
    alt: "logo of c++",
  },
  css: {
    img: css,
    alt: "logo of css",
  },
  js: {
    img: js,
    alt: "logo of javascript",
  },
  html: {
    img: html,
    alt: "logo of html",
  },
  ts: {
    img: ts,
    alt: "logo of typescript",
  },
  python: {
    img: python,
    alt: "logo of python programming language",
  },
  java: {
    img: java,
    alt: "logo of java",
  },
  pygame: {
    img: pygame,
    alt: "logo of pygame library",
  },
  sfml: {
    img: sfml,
    alt: "logo of SFML library",
  },
  tensflow: {
    img: tensflow,
    alt: "logo of tensorflow",
  },
  pandas: {
    img: pandas,
    alt: "logo of pandas library",
  },
  flask: {
    img: flask,
    alt: "logo of flask framework",
  },
  openai: {
    img: openai,
    alt: "logo of openai",
    invert: true,
  },
};

export { MiscIcons, CodeIcons, SocialIcons };
export type { Img };
