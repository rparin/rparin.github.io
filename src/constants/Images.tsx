import landBgR from "../assets/Landing/Red.svg";
import landBgP from "../assets/Landing/Purple.svg";
import landCode from "../assets/Landing/CodePy.webp";
import AboutCode from "../assets/About/CodeE.webp";
import AboutFood from "../assets/About/FoodE.webp";
import AboutGame from "../assets/About/LaptopE.webp";
import AboutBook from "../assets/About/BookE.webp";

const HomeImages = {
  bgR: {
    img: landBgR,
    alt: "Landing page background",
  },
  bgP: {
    img: landBgP,
    alt: "Landing page background",
  },
  code: {
    img: landCode,
    alt: "Landing page code background overlay",
  },
};

const AboutImages = {
  code: {
    img: AboutCode,
    alt: "Background image of study room",
  },
  book: {
    img: AboutBook,
    alt: "Background image of book in forest",
  },
  food: {
    img: AboutFood,
    alt: "Background image of food in cafe",
  },
  game: {
    img: AboutGame,
    alt: "Background image of laptop in urban city",
  },
};

export { HomeImages, AboutImages };
