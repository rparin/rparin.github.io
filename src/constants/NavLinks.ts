const NavItems = {
  projects: {
    id: "projects",
    title: "Projects",
    url: undefined,
  },
  about: {
    id: "about",
    title: "About",
    url: undefined,
  },
  contact: {
    id: "contact",
    title: "Contact",
    url: undefined,
  },
};

const NavHome = {
  id: "home",
};

const Resume = {
  id: "resume",
  title: "Resume",
  url: "/Ralph_Parin_Resume.pdf",
};

export { NavItems, NavHome, Resume };
