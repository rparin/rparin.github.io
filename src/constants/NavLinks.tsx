const NavItems = {
  projects: {
    id: "projects",
    title: "Projects",
  },
  about: {
    id: "about",
    title: "About",
  },
  contact: {
    id: "contact",
    title: "Contact",
  },
  resume: {
    id: "resume",
    title: "Resume",
    url: "https://raw.githubusercontent.com/rparin/rparin.github.io/main/public/Resume.pdf",
  },
};

const NavLinks = [
  {
    id: NavItems.projects.id,
    title: NavItems.projects.title,
  },
  {
    id: NavItems.about.id,
    title: NavItems.about.title,
  },
  {
    id: NavItems.contact.id,
    title: NavItems.contact.title,
  },
  {
    id: NavItems.resume.id,
    title: NavItems.resume.title,
    url: NavItems.resume.url,
  },
];

export { NavItems, NavLinks };
