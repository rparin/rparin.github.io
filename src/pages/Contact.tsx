import { AboutGen, Socials } from "@/constants/About";
import { SocialIcons } from "@/constants/Icons";
import Image from "next/image";

export default function Contact({ id }: { id: string }) {
  return (
    <section
      id={id}
      className="flex flex-col text-center before:content-[''] before:h-20 before:block mx-5 lg:mx-16">
      <div className="min-h-[45vh]">
        <h3 className="font-normal mt-10 text-2xl md:text-3xl">Get In Touch</h3>

        <div className="flex flex-wrap justify-center items-center gap-6 mt-10">
          <a title={SocialIcons.gmail.name} href={"mailto:" + AboutGen.email}>
            <Image
              className="max-h-[12vh] p-1 w-14 h-auto drop-shadow-xl hover:p-0"
              src={SocialIcons.gmail.img}
              alt={SocialIcons.gmail.alt}
            />
          </a>
          <a
            title={SocialIcons.linkedin.name}
            href={Socials.linkedIn}
            target="_blank"
            rel="noopener noreferrer">
            <Image
              className="max-h-[12vh] p-1 w-14  h-auto drop-shadow-xl hover:p-0"
              src={SocialIcons.linkedin.img}
              alt={SocialIcons.linkedin.alt}
            />
          </a>
          <a
            title={SocialIcons.github.name}
            href={Socials.github}
            target="_blank"
            rel="noopener noreferrer">
            <Image
              className="max-h-[12vh] p-1 w-14 h-auto drop-shadow-xl hover:p-0 dark:invert"
              src={SocialIcons.github.img}
              alt={SocialIcons.github.alt}
            />
          </a>
        </div>
      </div>

      <p className="mt-3 mb-3">Thanks for visiting!</p>
    </section>
  );
}
