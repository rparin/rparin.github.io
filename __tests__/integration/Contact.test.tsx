import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import { NavItems } from "@/constants/NavLinks";
import Contact from "@/pages/Contact";

describe("testing Contact Page Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(<Contact id={NavItems.contact.id} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<Contact id={NavItems.contact.id} />);
  expect(container).toMatchSnapshot();
});
