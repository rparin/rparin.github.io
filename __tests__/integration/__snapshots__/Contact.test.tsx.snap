// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <section
    class="mx-5 flex flex-col text-center before:block before:h-20 before:content-[''] lg:mx-16"
    id="contact"
  >
    <div
      class="min-h-[45vh]"
    >
      <h2
        class="mt-10 text-2xl font-normal md:text-3xl"
      >
        Get In Touch
      </h2>
      <div
        class="mt-10 flex flex-wrap items-center justify-center gap-6"
      >
        <a
          aria-label="Email Ralph"
          href="mailto:r.parin070@gmail.com"
        >
          <img
            alt="logo of Gmail"
            class="h-auto max-h-[12vh] w-14 p-1 drop-shadow-xl hover:p-0"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <a
          aria-label="Ralph's LinkedIn"
          href="https://www.linkedin.com/in/rparin"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="logo of LinkedIn"
            class="h-auto max-h-[12vh] w-14 p-1 drop-shadow-xl hover:p-0"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <a
          aria-label="Ralph's Github"
          href="https://github.com/rparin"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="logo of Github"
            class="h-auto max-h-[12vh] w-14 p-1 drop-shadow-xl hover:p-0 dark:invert"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
      </div>
    </div>
    <h3
      class="mb-3 mt-3 text-base font-normal"
    >
      Thanks for visiting!
    </h3>
  </section>
</div>
`;
